universe/Fleet.cpp

```diff
@@ -1391,13 +1391,101 @@ void Fleet::ShortenRouteToEndAtSystem(std::list<int>& travel_route, int last_sys
         travel_route.push_back(*m_travel_route.begin());
 }
 
-std::string Fleet::GenerateFleetName(const std::vector<int>& ship_ids, int new_fleet_id) {
+std::string Fleet::GenerateFleetName() {
     // TODO: Change returned name based on passed ship designs.  eg. return "colony fleet" if
     // ships are colony ships, or "battle fleet" if ships are armed.
-    if (new_fleet_id == INVALID_OBJECT_ID)
+    if (ID() == INVALID_OBJECT_ID)
         return UserString("NEW_FLEET_NAME_NO_NUMBER");
 
-    return boost::io::str(FlexibleFormat(UserString("NEW_FLEET_NAME")) % boost::lexical_cast<std::string>(new_fleet_id));
+    std::vector<TemporaryPtr<const Ship> > ships;
+    for (std::set<int>::iterator it = m_ships.begin(); it != m_ships.end(); ++it) {
+        if (TemporaryPtr<const Ship> ship = GetShip(*it)) {
+            ships.push_back(ship);
+        }
+    }
+
+    std::vector< TemporaryPtr<const Ship> >::iterator it = ships.begin();
+
+    // TODO C++11 replace all of these loops with std::all_of
+    bool all_monster(true);
+    while (it != ships.end()) {
```

Comment:
No technical reason, but two style reasons.

First.

`for` implies that something should be done for all elements.

`while` implies that something (in this case a check for a ship type) should be done for some elements until something happens (find a specific type).  The reader expects the break statement.  

And second.

The code that I wanted to write was like:

    `if (std::all_of(ship.begin(), ship.end(), [](const Ship & s){return s.IsMonster();}))
          return "Monster Fleet";`

`std::all_of` is only available in C++11 and it expands semantically into the while loop that I used.